remote-query/src/main/java/org/infinispan/tutorial/simple/remote/query/InfinispanRemoteQuery.java

```diff
@@ -0,0 +1,96 @@
+package org.infinispan.tutorial.simple.remote.query;
+
+import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.client.hotrod.configuration.ClientIntelligence;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+
+/**
+ * The Remote Query simple tutorial.
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ */
+public class InfinispanRemoteQuery {
+
+   public static void main(String[] args) throws Exception {
+      // Create a configuration for a locally-running server
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addServer().host("127.0.0.1")
+            .port(ConfigurationProperties.DEFAULT_HOTROD_PORT)
+            .marshaller(ProtoStreamMarshaller.class); // You need to specify the marshaller
+
+      // Connect to the server
+      RemoteCacheManager client = new RemoteCacheManager(builder.build());
+
+      // Get the persons cache, create it if needed with the default configuration
+      RemoteCache<String, Person> personsCache = client.administration().getOrCreateCache("people", "default");
+
+      // Create the persons dataset to be stored in the cache
+      Map<String, Person> persons = Stream.of(new Object[][]{
```

Comment:
Stream abuse. I'll call the Java police! :))

Reply:
okkkkk haha 👮